armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -202,48 +231,77 @@ def _conformKwargs(self, blueprint, matMods):
         return kwargs
 
     def _constructMaterial(self, blueprint, matMods):
-        nucsInProblem = blueprint.allNuclidesInProblem
-        # make material with defaults
-        mat = materials.resolveMaterialClassByName(self.material)()
-
-        if self.isotopics is not None:
-            # Apply custom isotopics before processing input mods so
-            # the input mods have the final word
-            blueprint.customIsotopics.apply(mat, self.isotopics)
-
-        # add mass fraction custom isotopics info, since some material modifications need to see them
-        # e.g. in the base Material.applyInputParams
-        matMods.update(
-            {
-                "customIsotopics": {
-                    k: v.massFracs for k, v in blueprint.customIsotopics.items()
-                }
-            }
-        )
-        if len(matMods) > 1:
-            # don't apply if only customIsotopics is in there
-            try:
-                # update material with updated input params from blueprints file.
-                mat.applyInputParams(**matMods)
-            except TypeError:
-                # This component does not accept material modification inputs of the names passed in
-                # Keep going since the modification could work for another component
-                pass
-
-        expandElementals(mat, blueprint)
-
-        missing = set(mat.p.massFrac.keys()).difference(nucsInProblem)
+        return constructMaterial(self.material, blueprint, matMods, self.isotopics)
```

Comment:
I'm sure you did this for a reason (you needed to call constructMaterial from elsewhere, but I'm wondering if it'd be best to just delete this method and call the new `constructMaterial` function directly from whoever calls this private method?

Reply:
Excellent suggestion